### DISCUSSION TITLE
How use useAsyncData only in event handlers?

### DISCUSSION THREAD

**Discussion by cawa-93** [nuxt/nuxt#17553] [Questions]
useAsyncData and useFetch have a very nice interface. But I would like to use them not to load previous display data, but for some event handlers, such as sending a form
```vue
<script setup lang="ts">
const {data, error, pending, send} = useFetch('/api/point')

const onSubmit = () => {
  // Trigger $fetch
  send({ method:'POST', body: {/* ... */} })
}
</script>

<template>
<form @submit.prevent="onSubmit">
  <span v-if="pending">Wait. Data is being sent.</span>
  <span v-if="error">Error: {{ error }}</span>
  <span v-if="data">Data has been sent.</span>

  <!-- Rest form -->
</form>
</template>
```
However, now, useFetch  trigger api request when the component is loaded, and there seems to be no other helper that offers the same API, but trigger $fetch manually.

**Comment by TheAlexLichter** [comment]
You should not. Please use `$fetch` instead ☺️

**Reply by cawa-93** [reply]
In this case I forced to re-create `useAsyncData` api (manage loading states, pending and errors) by hands 🤷🏻‍♂️
